src/client/consumer.rs

```diff
@@ -268,16 +350,20 @@ mod tests {
     struct MockFetchInner {
         batch_sizes: Vec<usize>,
         stream: mpsc::Receiver<Record>,
+        next_err: Option<Error>,
         buffer: Vec<Record>,
+        range: (i64, i64),
```

Comment:
The interaction with the offset tracking in `buffer` is perhaps a little funky. Perhaps this might be easier to understand if these were tracked automatically, and we added a delete records API... I dunno :sweat_smile:

Reply:
You mean you wanna mock a full Kafka broker? :wink: